packages/axios-odata-client/src/AxiosODataClient.ts

```diff
@@ -104,24 +104,19 @@ export class AxiosODataClient implements ODataClient<AxiosRequestConfig> {
     });
   }
 
-  private convertError(error: Error): Error | RequestError {
+  private convertError(error: Error): AxiosODataClientError {
     if ((error as AxiosError).isAxiosError) {
       const axiosError: AxiosError = error as AxiosError;
 
-      const status = axiosError.response != null ? axiosError.response.status : 0;
       const message = this.getErrorMessage(axiosError);
 
-      return Object.assign(new Error(), {
-        isRequestError: true,
-        stack: axiosError.stack,
-        message,
-        canceled: axios.isCancel(axiosError),
-        code: axiosError.code,
-        status,
-        data: axiosError.response ? axiosError.response.data : undefined,
-      });
+      if (message) {
```

Comment:
I'm really not sure if this distinction makes sense. Well, at least based on the message....
If the server implements error messages wrong, then we would show an Axios Error now.

I would suggest to go with a default error message in this case, something like "Failed without message!" or something along this line.